src/client-side-metrics/exporter.ts

```diff
@@ -0,0 +1,409 @@
+      'monitored_resource.zone': string;
+    };
+  };
+  scopeMetrics: {
+    scope: {
+      name: string;
+      version: string;
+    };
+    metrics: (CounterMetric | DistributionMetric)[];
+  }[];
+}
+
+/**
+ * Type guard function to determine if a given value is a counter value (a number).
+ *
+ * This function checks if a value, which could be either a `DistributionValue`
+ * object or a `number`, is specifically a `number`. This is used to differentiate
+ * between counter metrics (which have numeric values) and distribution metrics
+ * (which have more complex, object-based values).
+ *
+ */
+function isCounterValue(value: DistributionValue | number): value is number {
+  return typeof value === 'number';
+}
+
+/**
+ * Converts OpenTelemetry metrics data into a format suitable for the Google Cloud
+ * Monitoring API's `createTimeSeries` method.
+ *
+ * This function transforms the structured metrics data, including resource and
+ * metric attributes, data points, and aggregation information, into an object
+ * that conforms to the expected request format of the Cloud Monitoring API.
+ *
+ * @param {ExportInput} exportArgs - The OpenTelemetry metrics data to be converted. This
+ *   object contains resource attributes, scope information, and a list of
+ *   metrics with their associated data points.
+ *
+ * @returns An object representing a `CreateTimeSeriesRequest`, ready for sending
+ *   to the Google Cloud Monitoring API. This object contains the project name
+ *   and an array of time series data points, formatted for ingestion by
+ *   Cloud Monitoring.
+ *
+ * @throws Will throw an error if there are issues converting the data.
+ *
+ * @remarks
+ *   The output format is specific to the Cloud Monitoring API and involves
+ *   mapping OpenTelemetry concepts to Cloud Monitoring's data model, including:
+ *   - Mapping resource attributes to resource labels.
+ *   - Mapping metric attributes to metric labels.
+ *   - Handling different metric types (counter, distribution).
+ *   - Converting data points to the correct structure, including start and end
+ *     times, values, and bucket information for distributions.
+ *
+ * @example
+ *   const exportInput: ExportInput = { ... }; // Example ExportInput object
+ *   const monitoringRequest = metricsToRequest(exportInput);
+ *   // monitoringRequest can now be used in monitoringClient.createTimeSeries(monitoringRequest)
+ *
+ *
+ */
+export function metricsToRequest(exportArgs: ExportInput) {
+  const timeSeriesArray = [];
+  const resourceLabels = {
+    cluster: exportArgs.resource._syncAttributes['monitored_resource.cluster'],
+    instance:
+      exportArgs.resource._syncAttributes['monitored_resource.instance_id'],
+    project_id:
+      exportArgs.resource._syncAttributes['monitored_resource.project_id'],
+    table: exportArgs.resource._syncAttributes['monitored_resource.table'],
+    zone: exportArgs.resource._syncAttributes['monitored_resource.zone'],
```

Comment:
I mentioned above, but seeing `._syncAttributes` seems a little sketchy to me. That doesn't seem to be part of the advertised API surface

Reply:
`_syncAttributes` aren't part of the API surface, but they actually exist when the `export` method is called as shown in the screenshot in one of my other comments.

We could be grouping these under the data point attributes instead, but I think in other comments we discussed either way will work?

Reply:
I actually ended up deciding to group these under data point attributes mainly because then it doesn't tightly couple the metrics handler to a table.